src/dlgProfilePreferences.cpp

```diff
@@ -2119,15 +2121,29 @@ void dlgProfilePreferences::saveMap()
         return;
     }
 
-    QString fileName =
-            QFileDialog::getSaveFileName(this, tr("Save Mudlet map"), mapSaveLoadDirectory(pHost), tr("Mudlet map (*.dat *.json);;", "Do not change the extension text (in braces) - it is needed programmatically!"));
-    if (fileName.isEmpty()) {
-        return;
-    }
+    auto datFilter = tr("Mudlet binary (*.dat)", "Do not change extensions (in braces) as they are used programmatically");
+    auto jsonFilter = tr("Mudlet JSON map (*.json)", "Do not change extensions (in braces) as they are used programmatically");
+    auto saveExtensions(QStringList() << datFilter << jsonFilter);
+
+    QFileDialog dialog(this);
+    dialog.setWindowTitle(tr("Save Mudlet map"));
+    dialog.setDirectory(mapSaveLoadDirectory(pHost));
+    dialog.setNameFilter(saveExtensions.join(";;"));
+    dialog.setAcceptMode(QFileDialog::AcceptSave);
+    dialog.setDefaultSuffix("dat");
+    connect(&dialog,  &QFileDialog::filterSelected, this, [&dialog, datFilter, jsonFilter](const QString& filter) {
+        if (filter == datFilter) {
+            dialog.setDefaultSuffix("dat");
```

Comment:
```suggestion
            dialog.setDefaultSuffix(QStringLiteral("dat"));
```
🤔 Though as this is a duplicated it is probably better to create something like `const QString ext_dat{QStringLiteral("dat")};` and use `ext_dat` in both places...

Reply:
How do you gauge that probability?

Reply:
I don't agree with this... let's not do this please. It's just more work and indirection for a still to be proven benefit.

Reply:
Let's do QStringLiteral("dat") here as suggested, but not const ext_dat.

Reply:
Fair enough - I can live with *two* copies - it is when it gets into lots (for some unspecified value of *lots*) that it makes me itch... 😜